### PR TITLE
Re-249: Test configure of components

### PR DESCRIPTION
**Pull request by icemac**
Open issues:

- [x] Update to batou 2.3+ (currently testing against the latest final release 2.2.x)
- [x] fix tests also running on pull requests against something else than master.
- [ ] fix broken test (I do not know how that code could run with batou 2)
- [x] run GHA tests for all PRs
- [x] fix merge conflicts

Fixes flyingcircusio/batou#249.

### CODE REVIEW

**Comment by icemac** [comment]
@ctheune I cannot assign you as a reviewer here. Could you please have a first look at my commits?

File: setup.py [hunk 1]
```diff
@@ -1,29 +1,30 @@
...
     return os.path.join(*names)
 
 
 setup(
     name='batou_ext',
-    version='0.1dev0',
+    version='2.3.dev0',
     install_requires=[
-        'batou>=2',
+        'batou>=2, <2.3',
```
Comment (ctheune) [inline setup.py:18]: why <2.3? Is this the status intended to be useful for 2.2?

**Review by ctheune — APPROVED** [review]
This looks great. Looking forward to getting GitHub Actions in here. :)

File: setup.py (hunk 1 above)
Comment (icemac) [reply setup.py:18]: Currently we test against the latest final version.
It is not possible to have a version which supports both 2.2 and 2.3 (so maybe 2.3 is actually a version 3) – Only after creating the PR I found out that I am on the 2.3 branch, so this has to be fixed.

**Comment by icemac** [comment]
@ctheune Could you please approve the GHA actions run. One test will be breaking – I have no idea how this can have run in the past. I'll leave it to you to fix or delete the code.

I think I am done with the preparation of this PR.

Next steps would be:

* Fix the broken test
* Merge and release https://github.com/flyingcircusio/batou/pull/265
* Update `tox.ini` to no longer do a check-out of batou
* Merge the PR

**Comment by icemac** [comment]
I was a bit too fast: After reading the diff again I made some final fixes with the result that https://github.com/flyingcircusio/batou/pull/265 is no longer needed here.

**Comment by ctheune** [comment]
Hmm. I haven't seen the `_required_params_` stuff before and I'm a bit surprised by mixing test and regular code stuff. Also, tests currently fail when building the GH actions.

**Comment by icemac** [comment]
@ctheune The tests fail because of a problem in the actual code, which existed before this PR - I do not know how to solve it. Could you please look into it and give me some guidance how to deal with this missing attribute `environment_`? (Is it a leftover from older batou versions? Could it be as easy as renaming it to `environment`? Is that code broken and should be deleted? …)

**Comment by sallner** [comment]
@ctheune I see a lot of work going on in the repo lately. Maybe it would be good to have this tests running and ensure some testability of the components?

**Pull request merged**